### PR TITLE
optimize mem-fs-editor usage

### PR DESCRIPTION
**Pull request by mshima**
Fixes https://github.com/jhipster/generator-jhipster/issues/20782.

---

Please make sure the below checklist is followed for Pull Requests.

- [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [ ] Tests are added where necessary
- [ ] The JDL part is updated if necessary
- [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) is updated if necessary
- [ ] Documentation is added/updated where necessary
- [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (below reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

File: generators/languages/generator.mjs [hunk 1]
```diff
@@ -328,13 +328,13 @@ export default class LanguagesGenerator extends BaseApplicationGenerator {
     const generator = _this || this;
     const prefix = this.fetchFromInstalledJHipster('languages/templates');
     const langJavaProp = languageToJavaLanguage(lang);
-    generator.renderTemplate(
+    generator.writeFile(
```
Comment (Tcharl) [inline generators/languages/generator.mjs:331]: Can it be renamed writeTemplate if it is able to interpolate templates?

File: package.json [hunk 2]
```diff
@@ -102,15 +102,15 @@
     "isbinaryfile": "5.0.0",
     "js-yaml": "4.1.0",
     "lodash": "4.17.21",
-    "mem-fs-editor": "9.5.0",
+    "mem-fs-editor": "github:mshima/mem-fs-editor#perf",
```
Comment (mraible) [inline package.json:105]: Seems phishy. 🤔

Comment (mshima) [reply package.json:105]: I will release a new mem-fs-editor with the fix.

Comment (mshima) [reply package.json:105]: Released.

File: generators/languages/generator.mjs (hunk 1 above)
Comment (mshima) [reply generators/languages/generator.mjs:331]: writeFile matches the writeFiles.
IMO writeFile is ok.

**Review by DanielFran — APPROVED** [review]

**Pull request merged**